test/gateway/request.js

```diff
@@ -84,3 +84,82 @@ test('sendRequest method rejects when response contains errors', async (t) => {
 
   t.end()
 })
+
+test('sendRequest method should accept useSecureParse flag and parse the response securely', async (t) => {
+  const app = fastify()
+  app.post('/', async (request, reply) => {
+    return '{" __proto__": { "foo": "bar" } }'
+  })
+
+  await app.listen(0)
+
+  const url = new URL(`http://localhost:${app.server.address().port}`)
+  const { request, close } = buildRequest({ url })
+  t.teardown(() => {
+    close()
+    return app.close()
+  })
+  const result = await
+  sendRequest(
+    request,
+    url,
+    true
+  )({
+    method: 'POST',
+    body: JSON.stringify({
+      query: `
+    query ServiceInfo {
+      _service {
+        sdl
+      }
+    }
+    `
```

Comment:
I'm not sure I understand why we're sending a request body. what are we doing with it?

Reply:
Actually nothing, I copied the previous test and forgot to remove it. Will do it now.

Reply:
ok but they I'm wondering, what are we testing here? the behavior you're testing is on a plain route, not a graphql route. I'm confused

Reply:
We're testing if the response is parsed correctly with useSecureParse flag set to true.
If you look at the response you'll see it's a stringified object containing __proto__ the goal is to make sure that the secure parsing still works.

Reply:
what I don't understand is why this behavior is affecting plain HTTP requests. this is mercurius, why are we implementing behaviors which affect requests that should not be affected outside of what mercurius does?

Reply:
The documentation is not very clear about this, but this article sheds a bit more light on the topic: https://www.nearform.com/blog/implementing-a-graphql-gateway-with-fastify/

The idea is that `mercurius` can work in a so called *"gateway mode"*. In that mode the server is connecting to different services, after it gets the  results of those services, it combines the output of all of them into one final schema. Now, part of the gateway is the `request.js` , that Matteo asked to be changed, and the reasoning behind is I think clear from the ticket.

Reply:
thanks, makes sense now!